### ISSUE TITLE
Porting to latest GameLift version

### ISSUE THREAD

**Issue by klaus7** [BoxtrotStudio/AmazonGameLift-ServerSdk-Java#1]
Hello and thanks for providing a java SDK for the GameLift Server API. I'd like to use it, but have some issues with it.

For testing I start the GameLiftLocal.jar with: `java -jar GameLiftLocal.jar -p 9080`

I then added to your test class [MyServerClass.java](https://github.com/BoxtrotStudio/amazon-gamelift-serversdk-java/blob/master/src/test/java/com/boxtrotstudio/examples/MyServerClass.java)
```java
    public static void main(String[] args) throws GameLiftError {
        new MyServerClass().start();
    }
```
... and started it.

I then get the following error: 
```
Exception in thread "main" java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:653)
	at java.util.ArrayList.set(ArrayList.java:444)
	at com.google.protobuf.LazyStringArrayList.set(LazyStringArrayList.java:143)
	at com.google.protobuf.LazyStringArrayList.set(LazyStringArrayList.java:65)
	at com.amazonaws.gameLift.protobuf.Sdk$ProcessReady$Builder.setLogPathsToUpload(Sdk.java:3952)
	at com.boxtrotstudio.aws.common.AuxProxyMessageSender.processReady(AuxProxyMessageSender.java:104)
	at com.boxtrotstudio.aws.ServerState.processReady(ServerState.java:68)
	at com.boxtrotstudio.aws.GameLiftServerAPI.processReady(GameLiftServerAPI.java:35)
	at com.boxtrotstudio.examples.MyServerClass.start(MyServerClass.java:70)
	at com.boxtrotstudio.examples.MyServerClass.main(MyServerClass.java:13)
```

Something with the log files. For testing I tried to remove the log files. 
This:
```java
ProcessParameters parameters = new ProcessParameters(port, logs);
```
... became this:
```java
ProcessParameters parameters = new ProcessParameters(port);
```
New error:
```
Exception in thread "main" com.boxtrotstudio.aws.common.GameLiftError: Connection to local agent could not be established.
	at com.boxtrotstudio.aws.common.AuxProxyMessageSender.emitEvent(AuxProxyMessageSender.java:203)
	at com.boxtrotstudio.aws.common.AuxProxyMessageSender.emitEvent(AuxProxyMessageSender.java:191)
	at com.boxtrotstudio.aws.common.AuxProxyMessageSender.processReady(AuxProxyMessageSender.java:107)
	at com.boxtrotstudio.aws.ServerState.processReady(ServerState.java:68)
	at com.boxtrotstudio.aws.GameLiftServerAPI.processReady(GameLiftServerAPI.java:35)
	at com.boxtrotstudio.examples.MyServerClass.start(MyServerClass.java:70)
	at com.boxtrotstudio.examples.MyServerClass.main(MyServerClass.java:13)
Caused by: java.util.concurrent.TimeoutException
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:434)
	at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:85)
	at com.boxtrotstudio.aws.common.AuxProxyMessageSender.emitEvent(AuxProxyMessageSender.java:201)
	... 6 more
```

I can see the server trying to connect to the GameLiftLocal Service, but it is not able to handle GameSessions. This is the output of the GameLiftLocal Log:

```
10:50:03,215  INFO || - [SDKListenerImpl] nioEventLoopGroup-3-2 - SDK connected: /127.0.0.1:57741
10:50:03,216  INFO || - [SDKListenerImpl] nioEventLoopGroup-3-2 - SDK pid is 17972, sdkVersion is 3.1.5 and sdkLanguage is Java
10:50:03,220  INFO || - [SDKListenerImpl] nioEventLoopGroup-3-2 - NOTE: Only SDK versions 3.1.5 and above are supported in GameLiftLocal!
```

Any help would be greatly appreciated.

Best regards,
Klaus

**Comment by ecp4224** [comment]
Hey, I just pushed a fix to this. I had to rebuild the Sdk protobuf so it lives in the proper package `com.amazon.whitewater.auxproxy.pbuffer` and updated the Sdk protobuf to the latest version. 

Since there was no where online where the `sdk.proto` file lived, I also generated one from the C# SDK and added it to this repo.

Give it another go, it should work with the LocalGameLift now. 

However, while investigating this issue I see that this port is missing a lot of new properties that have been added since the last commit. I'll be working on adding those missing properties soon.

I'll close this issue once you can confirm the new build works

**Comment by ecp4224** [comment]
Also noting here, the commit also fixes the `Exception in thread "main" java.lang.IndexOutOfBoundsException` error so adding log files to your `ProcessParameters` should be ok

**Comment by klaus7** [comment]
Thanks a lot, really awesome that you could help out so fast. 👍 

With the fix I now can get past the processReady() but another problem arises when trying to create a game session via the following command: (and a running GameLiftLocal instance)

```
$ aws gamelift create-game-session --endpoint-url http://localhost:9080 --maximum-player-session-
count 2 --fleet-id fleet-1238
```
I get the response:
```
{
    "GameSession": {
        "FleetId": "fleet-1238",
        "Status": "ACTIVATING",
        "Port": 7777,
        "GameSessionId": "arn:aws:gamelift:local::gamesession/fleet-1238/gsess-f56e74f5-f1f0-4316-8422-245c5d13e187",
        "IpAddress": "127.0.0.1",
        "MaximumPlayerSessionCount": 2
    }
}
```
Which looks good at first, but the java server throws the following error:
```
com.google.protobuf.InvalidProtocolBufferException: Cannot find field: gameSession in message com.amazon.whitewater.auxproxy.pbuffer.GameSession
	at com.google.protobuf.util.JsonFormat$ParserImpl.mergeMessage(JsonFormat.java:1313)
	at com.google.protobuf.util.JsonFormat$ParserImpl.merge(JsonFormat.java:1273)
	at com.google.protobuf.util.JsonFormat$ParserImpl.merge(JsonFormat.java:1155)
	at com.google.protobuf.util.JsonFormat$Parser.merge(JsonFormat.java:338)
	at com.boxtrotstudio.aws.ServerState$3.run(ServerState.java:257)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```
... and won't invoke the whenGameSessionStarts() method.

Apart from the fix I'd also be really interested in how you went about updating the sdk protobuf to the latest version and also how you generated the sdk.proto file from the C# SDK? I actually wanted to do the same today, because I tried to fix it on my own, but I didn't come very far with it.

**Comment by ecp4224** [comment]
The C# server sdk has a `SDK.cs` file (not the `Sdk.cs` file) which has protobuf-net annotations that `Serializer.GetProto()` can use to generate a .proto file. I had to modify the generated output slightly so it used the correct package name and casing for variable names.

I believe the latest commit should fix that next error. I'm not sure what other errors you'll encounter as I don't have an aws setup to test with.  

I've renamed this issue so this issue focuses on porting to the latest version of GameLift.
